internal/server/iigointernal/utilities.go

```diff
@@ -0,0 +1,118 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/pkg/errors"
+)
+
+// PickUpRulesByVariable returns a list of rule_id's which are affected by certain variables
+func PickUpRulesByVariable(variableName string, ruleStore map[string]rules.RuleMatrix) ([]string, error) {
+	var Rules []string
+	if _, ok := rules.VariableMap[variableName]; ok {
+		for k, v := range ruleStore {
+			_, err := searchForStringInArray(variableName, v.RequiredVariables)
+			if err != nil {
+				Rules = append(Rules, k)
+			}
+		}
+		return Rules, nil
+	} else {
+		return []string{}, errors.Errorf("Variable name '%v' was not found in the variable cache", variableName)
+	}
+}
+
+func searchForStringInArray(val string, array []string) (int, error) {
+	for i, v := range array {
+		if v == val {
+			return i, nil
+		}
+	}
+	return 0, errors.Errorf("Not found")
+}
+
+type DataPacket struct {
+	integerData int
+	textData    string
+	booleanData bool
+}
+
+type Communication struct {
+	recipient int
+	sender    int
+	data      map[int]DataPacket
+}
+
+func broadcastToAllIslands(sender int, data map[int]DataPacket) {
+	islandsAlive := rules.VariableMap["islands_alive"]
+	for _, v := range islandsAlive.Values {
+		communicateWithIslands(int(v), sender, data)
+	}
+}
+
+func dataPacketToCommunication(d *DataPacket) baseclient.Communication {
```

Comment:
communication and datapacket are the same thing?

Reply:
Yup, sorry no idea why we chose this approach apart for the ensuing hilarity

Reply:
Removed now

Reply:
I had problem with cyclic imports, as datapacket is defined in utilities and receive function in baseclient takes datapacket as input. It could be that the massive refactoring fixed it, but I didn't notice